.azure_pipelines/performance_check/run_performance_check.py

```diff
@@ -0,0 +1,294 @@
+    model_list = ["hf_pytorch", "pytorch_compile", "onnx", "optimum_o1", "optimum_o2", "optimum_o3"]
+    if device == "gpu":
+        model_list.append("optimum_o4")
+    if device == "cpu":
+        model_list.append("optimum_dynamic_quantization")
+    metric_res = {}
+    config_json_path = cur_dir / "configs" / "perf.json"
+    for optimized_model in model_list:
+        metric_res[f"{optimized_model}"] = {}
+    metric_res["olive"] = {}
+    for i in range(test_num):
+        print(f"Start running {i} time...")
+        for optimized_model in model_list:
+            print(f"Start evaluating {optimized_model} model")
+            with open(config_json_path, "r") as fin:
+                olive_config = json.load(fin)
+                user_script_path = str(cur_dir / "user_scripts" / f"{model_name}.py")
+                hf_model_config = MODEL_NAME_TO_CONFIG_MAP[model_name]
+                if optimized_model == "onnx":
+                    olive_config["input_model"]["config"]["model_path"] = str(
+                        Path(model_root_path / optimized_model / "model.onnx")
+                    )
+                elif optimized_model == "optimum_dynamic_quantization":
+                    olive_config["input_model"]["config"]["model_path"] = str(
+                        Path(model_root_path / optimized_model / "model_quantized.onnx")
+                    )
+                elif optimized_model in ["optimum_o1", "optimum_o2", "optimum_o3", "optimum_o4"]:
+                    olive_config["input_model"]["config"]["model_path"] = str(
+                        Path(model_root_path / optimized_model / "model_optimized.onnx")
+                    )
+                elif optimized_model == "hf_pytorch":
+                    olive_config["input_model"]["type"] = "PyTorchModel"
+                    hf_config = {"hf_config": hf_model_config}
+                    olive_config["input_model"]["config"] = hf_config
+                elif optimized_model == "pytorch_compile":
+                    olive_config["input_model"]["type"] = "PyTorchModel"
+                    olive_config["input_model"]["config"]["model_script"] = user_script_path
+                    olive_config["input_model"]["config"]["model_loader"] = "torch_complied_model"
+
+                olive_config["systems"]["local_system"]["config"]["accelerators"] = (
+                    ["cpu"] if device == "cpu" else ["gpu"]
+                )
+                olive_config["engine"]["cache_dir"] = str(Path(model_root_path / optimized_model / "cache"))
+                olive_config["engine"]["output_dir"] = str(Path(model_root_path / optimized_model / "output"))
+                olive_config["engine"]["execution_providers"] = (
+                    ["CPUExecutionProvider"] if device == "cpu" else ["CUDAExecutionProvider"]
+                )
+                olive_config["evaluators"]["common_evaluator"]["metrics"].append(ACC_METRIC)
```

Comment:
Thanks for the changes!
This is not a big issue currently but maybe it is safer if we deepcopy `hf_config`, `ACC_METRIC` and `LATENCY_METRIC` so that we are using fresh values for each iteration? Since these values are passed by reference and updated by the workflow run. 
It might be hard to debug in the future if the workflow run/validation makes unexpected changes to these values.

Reply:
Sounds good to me. I'll copy metrics here.